testbed/datasenders/fluent.go

```diff
@@ -0,0 +1,160 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datasenders
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/fluent/fluent-logger-golang/fluent"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/testbed/testbed"
+)
+
+const (
+	fluentSplunkHECTestcseDatafileVar = "FLUENT_SPLUNKHEC_TESTCASE_DATA_FILE"
+	fluentPortVar                     = "FLUENT_SPLUNKHEC_FLUENT_PORT"
```

Comment:
Why are we coupling Fluent Forward data sender and Splunk? Can we keep Fluent Forward data sender independent from other formats?

Reply:
I was hoping we could use this test to give us a measure of performance while using `fluentforward` receiver with the `splunk_hec` exporter and having the test here would give us an easy way to reproduce results. Please let me know if it makes more sense to have such e2e tests some place else.

Reply:
I might have misunderstood your concern originally. @tigrannajaryan were you in this case just referring to the naming of the env variables being dependent on other formats? (and not the e2e test itself). If so, these case can be renamed.

Reply:
Updated the envs btw.

Reply:
Yes, I wonder if having "HEC" or "SPLUNK" in the name of the variable or the value is intentional. How and why is this related to HEC and Splunk? This is unexpected, since FluentForward is not related to HEC or Splunk in any way.